### PR TITLE
Numeric type options

### LINKED ISSUE #377: Handle `NUMERIC` data type in the backend.
This issue is to ensure that Mathesar can handle the Postgres `NUMERIC` data type.

As part of this issue, we need to ensure that:
- Users can use the API to change a column to `NUMERIC` if it's possible to do so.
- Users can set and change the `precision` and `scale` of a given `NUMERIC` column via a `type_options` field in the API.
- Automatic type inference during file import suggests `NUMERIC` when it makes sense to do so. 

**Additional Context** 
- See PR #368 for code that calculates which types are allowed for a column.

### PR DESCRIPTION
**Pull request by mathemancer**
Fixes #377 

This PR adds functionality to add `NUMERIC` type options (i.e., `scale` and `precision`) when creating, or altering, or previewing columns.  The options are used as arguments to the underlying SQL type.  

**Technical details**

The main big changes are actually the framework for using options (i.e., arguments) with types at all.  The type options are available upon reflection, and visible in the API.  To see the type _without_ arguments for a column, use the `MathesarColumn` class, and use the `plain_type` attr.  Now, when doing things with columns, use a new `"type_options"` key in the submitted JSON.  The value with this key should be a JSON object with either a `"precision"` key, or both `"precision"` and `"scale"` keys.

As a bonus, this PR removes `blah.html`, which was added by accident to the repo.

**Screenshots**

## Checklist
 
- [X] My pull request has a descriptive title (not a vague title like `Update
  index.md`).
- [X] My pull request targets the `master` branch of the repository
- [X] My commit messages follow [best practices][best_practices].
- [X] My code follows the established code style of the repository.
- [X] I added tests for the changes I made (if applicable).
- [ ] I added or updated documentation (if applicable).
- [X] I tried running the project locally and verified that there are no
  visible errors.

[best_practices]:https://gist.github.com/robertpainsi/b632364184e70900af4ab688decf6f53

## Developer Certificate of Origin

<details>
<summary>Developer Certificate of Origin</summary>

```
Developer Certificate of Origin
Version 1.1

Copyright (C) 2004, 2006 The Linux Foundation and its contributors.
1 Letterman Drive
Suite D4700
San Francisco, CA, 94129

Everyone is permitted to copy and distribute verbatim copies of this
license document, but changing it is not allowed.


Developer's Certificate of Origin 1.1

By making a contribution to this project, I certify that:

(a) The contribution was created in whole or in part by me and I
    have the right to submit it under the open source license
    indicated in the file; or

(b) The contribution is based upon previous work that, to the best
    of my knowledge, is covered under an appropriate open source
    license and I have the right under that license to submit that
    work with modifications, whether created in whole or in part
    by me, under the same open source license (unless I am
    permitted to submit under a different license), as indicated
    in the file; or

(c) The contribution was provided directly to me by some other
    person who certified (a), (b) or (c) and I have not modified
    it.

(d) I understand and agree that this project and the contribution
    are public and that a record of the contribution (including all
    personal information I submit with it, including my sign-off) is
    maintained indefinitely and may be redistributed consistent with
    this project or the open source license(s) involved.
```

</details>

### CODE REVIEW

File: db/columns.py [hunk 1]
```diff
@@ -166,15 +183,21 @@ def create_column(engine, table_oid, column_data):
...
-        column_data[NAME], sa_type, nullable=column_nullable,
+        column_data[NAME], sa_type(**column_type_options), nullable=column_nullable,
     )
     with engine.begin() as conn:
         ctx = MigrationContext.configure(conn)
         op = Operations(ctx)
         op.add_column(table.name, column, schema=table.schema)
-    return tables.reflect_table_from_oid(table_oid, engine).columns[column_data[NAME]]
+
+    new_column = MathesarColumn.from_column(
```
Comment (eito-fis) [inline db/columns.py:196]: Could we use `get_mathesar_column_with_engine` here?

**Review by eito-fis — COMMENTED** [review]
Not very familiar with types yet, but overall LGTM. Just one small nitpick.

File: db/columns.py (hunk 1 above)
Comment (mathemancer) [reply db/columns.py:196]: Yep, good eye.

**Review by kgodey — APPROVED** [review]
Looks good to me! I tested the API locally and everything seems to work well. I think the type options code looks great, it's very readable and seems easy to extend.

Also, thanks for removing `blah.html`, that was my fault.

**Pull request merged**